### PR TITLE
updated asc reader

### PR DESCRIPTION
**Pull request by mgeplf**
* Use BlockNeuronBuilder to simplify construction, and speed it up
* add more unwanted sections based on corpus of .asc files
* handles spines correctly

### CODE REVIEW

File: neurom/io/neurolucida.py [hunk 1]
```diff
@@ -105,165 +88,154 @@ def _get_tokens(morph_fd):
                 yield token
 
 
+def _consume_until_balanced_paren(token_iter, opening_count=1):
+    '''Consume tokens until a opening_count close parens, taking into account balanced pairs'''
+    opening_count = 1
+    for token in token_iter:
+        if token == ')':
+            opening_count -= 1
```
Comment (arsenius7) [inline neurom/io/neurolucida.py]: not checking for negative values? ^^

File: neurom/io/neurolucida.py [hunk 2]
```diff
@@ -105,165 +88,154 @@ def _get_tokens(morph_fd):
                 yield token
 
 
+def _consume_until_balanced_paren(token_iter, opening_count=1):
+    '''Consume tokens until a opening_count close parens, taking into account balanced pairs'''
+    opening_count = 1
```
Comment (arsenius7) [inline neurom/io/neurolucida.py]: what's the point of passing the argument value then?

File: neurom/io/datawrapper.py [hunk 3]
```diff
@@ -271,3 +271,10 @@ def get_datawrapper(self, file_format='BlockNeuronBuilder', data_wrapper=DataWra
         self._check_consistency()
         datablock, sections = self._make_datablock()
         return data_wrapper(datablock, file_format, sections)
+
+    def __str__(self):
+        '''pretty print neuron'''
+        ret = []
+        for section_id, section in self.sections.items():
+            ret.append(str(section_id) + ' ' + str(section))
```
Comment (arsenius7) [inline neurom/io/datawrapper.py]: No love for list comprehensions? Weird :)
["{0} {1}".format(kv) for kv in dict.items()]

**Comment by asanin-epfl** [comment]
closing because NeuroM v2 uses MorphIO as its reader.

**Pull request closed without merging**